tensorflow_datasets/structured/adult.py

```diff
@@ -0,0 +1,208 @@
+    "marital_status", [
+        "Married-civ-spouse", "Divorced", "Married-spouse-absent",
+        "Never-married", "Separated", "Married-AF-spouse", "Widowed"
+    ])
+
+def relationship_cat(d):
+  return tf.feature_column.categorical_column_with_vocabulary_list(
+    "relationship", [
+        "Husband", "Not-in-family", "Wife", "Own-child", "Unmarried",
+        "Other-relative"
+    ])
+
+def workclass_cat(d):
+  return tf.feature_column.categorical_column_with_vocabulary_list(
+    "workclass", [
+        "Self-emp-not-inc", "Private", "State-gov", "Federal-gov",
+        "Local-gov", "?", "Self-emp-inc", "Without-pay", "Never-worked"
+    ])
+
+def cat_hash(a):
+  return tf.feature_column.categorical_column_with_hash_bucket(a, hash_bucket_size=1000)
+
+def numeric(a):
+  return tf.feature_column.numeric_column(a)
+
+FEATURE_DICT = collections.OrderedDict([
+    ("gender", (tf.string, gender_cat)),
+    ("education", (tf.string, education_cat)),
+    ("marital_status", (tf.string, marital_status_cat)),
+    ("relationship", (tf.string, relationship_cat)),
+    ("workclass", (tf.string, workclass_cat)),
+    ("occupation", (tf.string, cat_hash)),
+    ("native_country", (tf.string, cat_hash)),
+    ("age", (tf.float32, numeric)),
+    ("education_num", (tf.float32, numeric)),
+    ("capital_gain", (tf.float32, numeric)),
+    ("capital_loss", (tf.float32, numeric)),
+    ("hours_per_week", (tf.float32, numeric))
+])
+
+_INCOME_DICT = {'>50K' : 'below', '<=50K' : 'above'}
+
+_URL_TRAIN = 'https://archive.ics.uci.edu/ml/machine-learning-databases/adult/adult.data'
+
+_URL_TEST = 'https://archive.ics.uci.edu/ml/machine-learning-databases/adult/adult.test'
+
+class Adult(tfds.core.GeneratorBasedBuilder):
+
+  """Prediction task is to determine whether a person makes over 50K a year."""
+ 
+  VERSION = tfds.core.Version('0.1.0')
+
+  def _info(self):
+    return tfds.core.DatasetInfo(
+        builder=self,
+        description=_DESCRIPTION,
+        features=tfds.features.FeaturesDict({
+            "income": tfds.features.ClassLabel(names=["below", "above"]),
+            "features": {name: dtype
+                         for name, (dtype, func) in FEATURE_DICT.items()}
+        }),
+        supervised_keys=("features", "income"),
+        homepage="https://archive.ics.uci.edu/ml/datasets/adult",
+        citation=_CITATION
+        )
+
+  def _split_generators(self, dl_manager):
+    
+    train = dl_manager.download(_URL_TRAIN)
+    test = dl_manager.download(_URL_TEST)
+    return [
+        tfds.core.SplitGenerator(
+            name=tfds.Split.TRAIN,
+            num_shards=1,
+            gen_kwargs={
+                "file_path": train
+            }),
+        
+        tfds.core.SplitGenerator(
+            name=tfds.Split.TEST,
+            num_shards=1,
+            gen_kwargs={
+                "file_path": test
+            }),
+            ]
+
+
+  def _generate_examples(self):
```

Comment:
Have you tested this? You should pass `file_path` as an argument in `_generate_examples`.